proxy/src/main/java/com/wavefront/agent/AbstractAgent.java

```diff
@@ -198,6 +197,10 @@ private void postProcessConfig() {
       throw new IllegalArgumentException("hostname cannot be blank! Please correct your configuration settings.");
     }
 
+    if (StringUtils.isBlank(proxyConfig.getProxyname().trim())) {
+      throw new IllegalArgumentException("proxyname cannot be blank! Please correct your configuration settings.");
```

Comment:
no need for this, if the `proxyname` is not set, use `getLocalHostName();` as default value

Reply:
This is in the case if proxyname is set to empty string in config file (`proxyname=`). If proxyname is not provided, it defaults to hostname